app/Policies/UserPolicy.php

```diff
@@ -37,30 +21,33 @@ public function before(?User $user, $ability)
 	 */
 	public function isAdmin(?User $user): bool
 	{
-		return $user?->id === 0;
+		return $user?->may_administrate === true;
 	}
 
 	/**
-	 * This defines if user can edit their settings.
+	 * Perform pre-authorization checks.
 	 *
-	 * @param User $user
+	 * @param User|null $user
+	 * @param string    $ability
 	 *
-	 * @return bool
+	 * @return void|bool
 	 */
-	public function canEditSettings(User $user): bool
+	public function before(?User $user, $ability)
 	{
-		return !$user->is_locked;
+		if ($this->isAdmin($user)) {
+			return true;
+		}
 	}
 
 	/**
-	 * This defines if user has upload rights.
+	 * This defines if user can edit their settings.
 	 *
 	 * @param User $user
 	 *
 	 * @return bool
 	 */
-	public function canUpload(User $user): bool
+	public function canEditOwnSettings(User $user): bool
 	{
-		return $user->may_upload;
+		return $user->may_edit_own_settings;
 	}
```

Comment:
I still believe we should have

```php
public function canEdit(User $user, User $otherUser): bool {
    return $user->may_edit_own_settings && $user->id === $otherUser->id;
}
```

Moreover, we probably need a right for creating/deleting users like this

```php
public function canCreateOrDelete(User $user): bool {
    // Note, the administrator is already handled in the `before()` method and every one else is not allowed to create/delete users.
    return false;
}
```